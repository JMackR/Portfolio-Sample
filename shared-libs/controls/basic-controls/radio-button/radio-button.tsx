import React from 'react'
import { Flex, Margin } from '@sample/layout-controls'
import { Checkbox, Text } from '@sample/basic-controls'
import styled from 'styled-components'
import { useBorder, useColorForBackgroundColor, useColorForTextColor } from '@sample/themes'

const RadioButtonStyled = styled.button`
  background-color: ${(props) =>
    props.theme.selected ? useColorForBackgroundColor('highlight1') : useColorForBackgroundColor('background1')};
  padding: 16px;
  border: ${(props) => (props.theme.selected ? '1px solid transparent' : `1px solid ${useColorForTextColor('primary7')}`)};
  text-align: start;
  border-radius: ${(props) => props.theme.borderRadius}px;
  cursor: pointer;

  &:hover {
    background-color: ${() => useColorForBackgroundColor('highlight2')};
  }
`

export interface RadioButtonProps {
  selected: boolean
  text: string
  onClick: () => void
}

export const RadioButton = ({ selected, text, onClick }: RadioButtonProps) => {
  const { baseBorder } = useBorder()

  return (
    <RadioButtonStyled
      theme={{
        borderRadius: baseBorder.cornerRadius.small,
        selected: selected,
      }}
      onClick={onClick}
    >
      <Flex crossAxisDistribution={'center'}>
        <Checkbox
          checked={selected}
          onChange={() => {}}
          checkedBackgroundColor={'background2'}
          circle={true}
          borderColor={useColorForTextColor('primary7')}
        />
        <Margin marginLeftStep={2}>
          <Text text={text} textType={'primary3'} color={'primary3'} />
        </Margin>
      </Flex>
    </RadioButtonStyled>
  )
}
