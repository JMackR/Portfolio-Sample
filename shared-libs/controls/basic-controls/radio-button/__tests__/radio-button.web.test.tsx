import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { RadioButton, RadioButtonProps } from '../radio-button'

jest.mock('@sample/assets', () => ({
  CheckBoxSelected: () => ({
    SVG: () => <div>CheckBoxSelected</div>,
  }),
}))
jest.mock('../../image/svgs/svg.tsx')

describe('Radio Button', () => {
  let props: RadioButtonProps

  beforeEach(() => {
    props = {
      selected: false,
      text: 'Test Text',
      onClick: jest.fn(),
    }
  })

  it('should render the text', () => {
    const { getByText } = render(<RadioButton {...props} />)

    expect(getByText('Test Text')).toBeDefined()
  })

  describe('the user clicks the button', () => {
    it('should call on click', () => {
      const { getByRole } = render(<RadioButton {...props} />)

      const button = getByRole('button')

      fireEvent.click(button)

      expect(props.onClick).toHaveBeenCalledTimes(1)
    })
  })
})

describe('Radio Button Snapshot Tests', () => {
  let props: RadioButtonProps
  beforeEach(() => {
    props = {
      selected: false,
      text: 'Test Text',
      onClick: jest.fn(),
    }
  })

  describe('the button is not selected', () => {
    it('renders properly', () => {
      const tree = renderer.create(<RadioButton {...props} />)

      expect(tree).toMatchSnapshot()
    })
  })

  describe('the button is selected', () => {
    beforeEach(() => {
      props.selected = true
    })

    it('renders properly', () => {
      const tree = renderer.create(<RadioButton {...props} />)

      expect(tree).toMatchSnapshot()
    })
  })
})
