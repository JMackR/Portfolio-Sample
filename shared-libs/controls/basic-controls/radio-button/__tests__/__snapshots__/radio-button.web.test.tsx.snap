// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Button Snapshot Tests the button is not selected renders properly 1`] = `
.flex {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  flex-wrap: undefined;
  align-items: center;
  justify-content: flex-start;
  max-width: 100vw;
  width: undefined;
  height: undefined;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.margin {
  margin: undefined;
  margin-left: 8px;
  margin-right: undefined;
  margin-top: undefined;
  margin-bottom: undefined;
  display: flex;
  position: relative;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  max-width: undefined;
  width: undefined;
}

.text-textStyle {
  color: #23231E;
  font-size: 16px;
  font-weight: 700;
  font-family: ABCSocial-Bold, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
  text-decoration-line: undefined;
}

<button
  className="sc-kDDrLX kXedRd"
  onClick={[MockFunction]}
>
  <div
    className="flex"
  >
    <label>
      <div
        className="sc-iBkjds jxRPHA"
      >
        <input
          checked={false}
          className="sc-papXJ hTulOx"
          onChange={[Function]}
          type="checkbox"
        />
        <div
          checked={false}
          className="sc-jqUVSM haHvhk"
        >
          <div>
            SVG
          </div>
        </div>
      </div>
    </label>
    <div
      className="margin"
    >
      <div
        className="text-textStyle"
      >
        Test Text
      </div>
    </div>
  </div>
</button>
`;

exports[`Radio Button Snapshot Tests the button is selected renders properly 1`] = `
.flex {
  display: flex;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  flex-wrap: undefined;
  align-items: center;
  justify-content: flex-start;
  max-width: 100vw;
  width: undefined;
  height: undefined;
  position: relative;
  background-color: undefined;
  overflow: undefined;
}

.margin {
  margin: undefined;
  margin-left: 8px;
  margin-right: undefined;
  margin-top: undefined;
  margin-bottom: undefined;
  display: flex;
  position: relative;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  flex-direction: row;
  align-items: stretch;
  justify-content: flex-start;
  background-color: undefined;
  height: undefined;
  max-width: undefined;
  width: undefined;
}

.text-textStyle {
  color: #23231E;
  font-size: 16px;
  font-weight: 700;
  font-family: ABCSocial-Bold, sans-serif;
  display: inline;
  white-space: inherit;
  text-align: undefined;
  text-decoration-line: undefined;
}

<button
  className="sc-kDDrLX fEuRJY"
  onClick={[MockFunction]}
>
  <div
    className="flex"
  >
    <label>
      <div
        className="sc-iBkjds jxRPHA"
      >
        <input
          checked={true}
          className="sc-papXJ hTulOx"
          onChange={[Function]}
          type="checkbox"
        />
        <div
          checked={true}
          className="sc-jqUVSM enkiKM"
        >
          <div>
            SVG
          </div>
        </div>
      </div>
    </label>
    <div
      className="margin"
    >
      <div
        className="text-textStyle"
      >
        Test Text
      </div>
    </div>
  </div>
</button>
`;
